lib/http.js

```diff
@@ -81,17 +81,79 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
     cors(),
     // FIXME: add validation
     //validate({bodySchema: updateMeterBody}),
-    _createAuthorizeMiddleware(),
+    _addMeterRecordToRequest(),
+    cors(),
+    _createAuthorizeMiddleware({
+      getExpectedTarget({req}) {
+        // only support update meter config URL as expected target; controller
+        // is the controller of the meter
+        const baseUrl = `https://${req.get('host')}${routes.basePath}`;
+        const expectedTarget =
+          `${baseUrl}/${encodeURIComponent(req.params.meterId)}`;
+        return {expectedTarget};
+      },
+      getRootController({req}) {
+        // root controller for update meter config is always the controller of
+        // the meter
+        const {meter: {controller}} = req.meterRecord;
+        return controller;
+      }
+    }),
     asyncHandler(async (req, res) => {
       let {body: meter} = req;
 
       // update meter according to handler result
       ({meter} = await HANDLERS.update({meter}));
 
-      // FIXME: update meter using meters.update()
+      // specify the max number of retries when InvalidStateError is thrown
+      const maxRetries = 10;
+      let attempts = 0;
 
-      // FIXME: implement
-      res.send(503);
+      if(meter.id) {
+        // remove meter id to prevent overwriting the original id
+        delete meter.id;
+      }
+
+      // retry until attempts are exhausted
+      while(attempts < maxRetries) {
+        try {
+          // update the meter
+          const updatedMeter = {
+            ...req.meterRecord.meter,
+            ...meter,
```

Comment:
I think we need to be more careful here. Is there any reason for this meter service to allow a user to change the service ID associated with the meter? I would think that would be trouble. I suspect we need to be more careful about which fields are mutable in a meter.